### PR TITLE
server: keep retrying bootstrapping

### PR DESCRIPTION
**Pull request by sttts**
A single error could stop server bootstrapping.

### CODE REVIEW

File: pkg/server/server.go [hunk 1]
```diff
@@ -115,19 +115,30 @@ func (s *Server) Run(ctx context.Context) error {
...
 			// nolint:nilerr
 			return nil // don't klog.Fatal. This only happens when context is cancelled.
 		}
 		klog.Infof("Finished bootstrapping system CRDs")
 
-		if err := configshard.Bootstrap(goContext(ctx), s.KcpClusterClient.Cluster(systemShardCluster)); err != nil {
+		if err := wait.PollInfiniteWithContext(goContext(ctx), time.Second, func(ctx context.Context) (bool, error) {
+			if err := configshard.Bootstrap(ctx, s.KcpClusterClient.Cluster(systemShardCluster)); err != nil {
+				klog.Errorf("Failed to bootstrap the shard workspace: %v", err)
+				return false, nil // keep trzing
```
Comment (p0lyn0mial) [inline pkg/server/server.go:138]: nit: a typo

File: pkg/server/server.go [hunk 2]
```diff
@@ -115,19 +115,30 @@ func (s *Server) Run(ctx context.Context) error {
 
 		klog.Infof("Finished start kube informers")
 
-		if err := systemcrds.Bootstrap(
-			goContext(ctx),
-			s.ApiExtensionsClusterClient.Cluster(SystemCRDLogicalCluster),
-			s.ApiExtensionsClusterClient.Cluster(SystemCRDLogicalCluster).Discovery(),
-			s.DynamicClusterClient.Cluster(SystemCRDLogicalCluster),
-		); err != nil {
+		if err := wait.PollInfiniteWithContext(goContext(ctx), time.Second, func(ctx context.Context) (bool, error) {
```
Comment (p0lyn0mial) [inline pkg/server/server.go:118]: I think 1s is okay

File: pkg/server/server.go [hunk 3]
```diff
@@ -115,19 +115,30 @@ func (s *Server) Run(ctx context.Context) error {
...
+			if err := systemcrds.Bootstrap(ctx,
+				s.ApiExtensionsClusterClient.Cluster(SystemCRDLogicalCluster),
+				s.ApiExtensionsClusterClient.Cluster(SystemCRDLogicalCluster).Discovery(),
+				s.DynamicClusterClient.Cluster(SystemCRDLogicalCluster),
+			); err != nil {
+				klog.Errorf("failed to bootstrap system CRDs: %v", err)
+				return false, nil // keep trzing
```
Comment (p0lyn0mial) [inline pkg/server/server.go:125]: nit: a typo here as well

File: pkg/server/server.go [hunk 4]
```diff
@@ -115,19 +115,30 @@ func (s *Server) Run(ctx context.Context) error {
...
-			s.ApiExtensionsClusterClient.Cluster(SystemCRDLogicalCluster).Discovery(),
-			s.DynamicClusterClient.Cluster(SystemCRDLogicalCluster),
-		); err != nil {
+		if err := wait.PollInfiniteWithContext(goContext(ctx), time.Second, func(ctx context.Context) (bool, error) {
+			if err := systemcrds.Bootstrap(ctx,
+				s.ApiExtensionsClusterClient.Cluster(SystemCRDLogicalCluster),
+				s.ApiExtensionsClusterClient.Cluster(SystemCRDLogicalCluster).Discovery(),
+				s.DynamicClusterClient.Cluster(SystemCRDLogicalCluster),
+			); err != nil {
+				klog.Errorf("failed to bootstrap system CRDs: %v", err)
```
Comment (p0lyn0mial) [inline pkg/server/server.go:124]: when and how can this actually fail?

Comment (sttts) [reply pkg/server/server.go:124]: network error, wrong manifests, any kind of 50x

File: pkg/server/server.go (hunk 3 above)
Comment (sttts) [reply pkg/server/server.go]: fixed

File: pkg/server/server.go (hunk 1 above)
Comment (sttts) [reply pkg/server/server.go]: fixed

File: pkg/server/server.go (hunk 4 above)
Comment (p0lyn0mial) [reply pkg/server/server.go:124]: > a network error

aren't we using the local client?

> wrong manifest 

this can never succeed

**Pull request merged**